File: 42842.cpp
// 카펫

#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int red) {
    vector<int> answer;
    int length = brown + red;
    for (int i = 3; i < length; i++) {
        if (length % i == 0) {
            int j = length / i;
            if (brown == (i * 2 ) + (j * 2) - 4) {
                answer.push_back(j);
                answer.push_back(i);
                break;
            }
        }
    }
    return answer;
}

File: 42842.py
# 카펫
# 2021-09-22

def solution(brown, yellow):
    answer = []
    for i in range(1, int(yellow ** 0.5) + 1):
        if not yellow % i:
            if brown == 2 * (i + yellow // i) + 4:
                answer = [i + 2, yellow // i + 2]
    return sorted(answer, reverse=True)

print(solution(10, 2))